docs/dns-policy.md

```diff
@@ -0,0 +1,497 @@
+      reason: Programmed
+      status: "True"
+      type: Programmed
+  listeners:
+    - attachedRoutes: 1
+      conditions: []
+      name: kind-mgc-workload-1.api
+      supportedKinds: []
+    - attachedRoutes: 1
+      conditions: []
+      name: kind-mgc-workload-2.api
+      supportedKinds: []        
+```
+
+The example DNSPolicy shown above would create a DNSRecord like the following:
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSRecord
+metadata:
+  creationTimestamp: "2023-07-24T19:09:56Z"
+  finalizers:
+    - kuadrant.io/dns-record
+  generation: 3
+  labels:
+    kuadrant.io/Gateway-uid: 0877f97c-f3a6-4f30-97f4-e0d7f25cc401
+    kuadrant.io/record-id: echo
+  name: echo.apps.hcpapps.net
+  namespace: multi-cluster-gateways
+  ownerReferences:
+    - apiVersion: gateway.networking.k8s.io/v1beta1
+      kind: Gateway
+      name: echo-app
+      uid: 0877f97c-f3a6-4f30-97f4-e0d7f25cc401
+    - apiVersion: kuadrant.io/v1alpha1
+      blockOwnerDeletion: true
+      controller: true
+      kind: ManagedZone
+      name: apps.hcpapps.net
+      uid: 26a06799-acff-476b-a1a3-c831fd19dcc7
+  resourceVersion: "25464"
+  uid: 365bf57f-10b4-42e8-a8e7-abb6dce93985
+spec:
+  endpoints:
+    - dnsName: 24osuu.lb-2903yb.echo.apps.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      targets:
+        - 172.32.202.1
+    - dnsName: default.lb-2903yb.echo.apps.hcpapps.net
+      providerSpecific:
+        - name: weight
+          value: "120"
+      recordTTL: 60
+      recordType: CNAME
+      setIdentifier: 24osuu.lb-2903yb.echo.apps.hcpapps.net
+      targets:
+        - 24osuu.lb-2903yb.echo.apps.hcpapps.net
+    - dnsName: default.lb-2903yb.echo.apps.hcpapps.net
+      providerSpecific:
+        - name: weight
+          value: "120"
+      recordTTL: 60
+      recordType: CNAME
+      setIdentifier: lrnse3.lb-2903yb.echo.apps.hcpapps.net
+      targets:
+        - lrnse3.lb-2903yb.echo.apps.hcpapps.net
+    - dnsName: echo.apps.hcpapps.net
+      recordTTL: 300
+      recordType: CNAME
+      targets:
+        - lb-2903yb.echo.apps.hcpapps.net
+    - dnsName: lb-2903yb.echo.apps.hcpapps.net
+      providerSpecific:
+        - name: geo-country-code
+          value: '*'
+      recordTTL: 300
+      recordType: CNAME
+      setIdentifier: default
+      targets:
+        - default.lb-2903yb.echo.apps.hcpapps.net
+    - dnsName: lrnse3.lb-2903yb.echo.apps.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      targets:
+        - 172.32.201.1
+  managedZone:
+    name: apps.hcpapps.net   
+```
+
+Which results in the following records being created in AWS Route53 (The provider we used in our example ManagedZone above):
+
+![aws-recordset-list-1](./images/dns-policy/aws-recordset-list-1.png)
+
+The listener hostname is now be resolvable through dns:
+
+```bash
+dig echo.apps.hcpapps.net +short
+lb-2903yb.echo.apps.hcpapps.net.
+default.lb-2903yb.echo.apps.hcpapps.net.
+lrnse3.lb-2903yb.echo.apps.hcpapps.net.
+172.32.201.1
```

Comment:
There are a lot of records created.
Is this a good place to explain why that's the case e.g. to future proof so we can do geo later if you want
It can be a little overwhelming trying to make sense of them all.

Reply:
yeah, I'll have a go at trying to explain the structure and the reason for it somewhere in here.

Reply:
maybe we link to the DNS structure doc?

Reply:
Yeah , I'm going to link to the proposal doc, it's got everything in it. We might want to consider going back and moving that to an actual doc rather than a proposal, but will link to that for now.